### PR TITLE
Check np for LifecycleComponent close

### PR DESCRIPTION
**Pull request by rudy2steiner**
Descriptions of the changes in this PR:

 Check NP of  BookKeeper Stream Storage Server LifecycleComponents

### Motivation

Fix issue #2505 
 
### Changes

(Describe: what changes you have made)

Master Issue: #2505

### CODE REVIEW

**Review by eolivelli — APPROVED** [review]
LGTM
@Ghatage  @sijie @rdhabalia @fpj PTAL

this is a good candidate for 4.12.2, we have a few little changes that are helping in improving the user experience, especially during "problems"

File: stream/server/src/main/java/org/apache/bookkeeper/stream/server/service/RegistrationStateService.java [hunk 1]
```diff
@@ -95,6 +95,10 @@ protected void doStart() {
 
     @Override
     protected void doStop() {
+        if (null == stateManager) {
+            log.warn("State Manager is null, no need to stop it.");
```
Comment (fpj) [inline stream/server/src/main/java/org/apache/bookkeeper/stream/server/service/RegistrationStateService.java:99]: A warn message for me indicates a potential action from the user. In this case, it is just information that the state manager is null, so no action is required, correct? If so, then it is an info message at best, if not debug.

Comment (rudy2steiner) [reply stream/server/src/main/java/org/apache/bookkeeper/stream/server/service/RegistrationStateService.java:99]: yes

**Comment by eolivelli** [comment]
@fpj 's has been addresses, I merged the patch

thank you @rudy2steiner

**Pull request merged**